packages/components/src/components/data-table/data-table.jsx

```diff
@@ -12,57 +12,50 @@ import ThemedScrollbars from '../themed-scrollbars';
       2. implement filtering per column
 */
 
-class DataTable extends React.PureComponent {
-    cache_ref = React.createRef();
-    is_dynamic_height = !this.props.getRowSize;
-    state = {
-        scrollTop: 0,
-        is_loading: true,
-    };
+const DataTable = ({
+    children,
+    className,
+    columns,
+    content_loader,
+    data_source,
+    footer,
+    getActionColumns,
+    getRowSize,
+    keyMapper,
+    onScroll,
+    ...other_props
+}) => {
+    const cache_ref = React.useRef();
+    const list_ref = React.useRef();
+    const is_dynamic_height = !getRowSize;
+    const [scroll_top, setScrollTop] = React.useState(0);
+    const [is_loading, setLoading] = React.useState(true);
 
-    componentDidMount() {
-        if (this.is_dynamic_height) {
-            this.cache_ref.current = new CellMeasurerCache({
+    React.useEffect(() => {
+        if (is_dynamic_height) {
+            cache_ref.current = new CellMeasurerCache({
                 fixedWidth: true,
                 keyMapper: row_index => {
-                    if (row_index < this.props.data_source.length)
-                        return this.props.keyMapper?.(this.props.data_source[row_index]) || row_index;
+                    if (row_index < data_source.length) return keyMapper?.(data_source[row_index]) || row_index;
                     return row_index;
                 },
             });
         }
-        this.setState({ is_loading: false });
-    }
+        setLoading(false);
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, []);
 
-    componentDidUpdate(prevProps) {
-        if (this.props.data_source !== prevProps.data_source || this.props.getRowSize !== prevProps.getRowSize) {
-            if (this.is_dynamic_height) {
-                this.list_ref.current?.recomputeGridSize(0);
-            }
-        }
-    }
+    React.useEffect(() => {
+        if (is_dynamic_height) list_ref.current?.recomputeGridSize(0);
+    }, [data_source, is_dynamic_height]);
 
-    handleScroll = ev => {
-        const { scrollTop } = ev.target;
-        this.setState({ scrollTop });
-        if (typeof this.props.onScroll === 'function') {
-            this.props.onScroll(ev);
-        }
+    const handleScroll = ev => {
+        setScrollTop(ev.target.scrollTop);
+        if (typeof onScroll === 'function') onScroll(ev);
     };
 
-    rowRenderer = ({ style, index, key, parent }) => {
-        const {
-            className,
-            columns,
-            content_loader,
-            data_source,
-            getActionColumns,
-            getRowAction,
-            id,
-            keyMapper,
-            passthrough,
-            preloaderCheck,
-        } = this.props;
+    const rowRenderer = ({ style, index, key, parent }) => {
+        const { getRowAction, id, passthrough, preloaderCheck } = other_props;
```

Comment:
any reason for separating component props and `other_props` in different places?

Reply:
Yeah better to not separate the props. Will fix it.